src/ray/gcs/gcs_server/gcs_placement_group_scheduler.h

```diff
@@ -149,43 +154,94 @@ class LeaseStatusTracker {
+  /// \param node_id Id of a node where prepare request is returned.
+  /// \param bundle Bundle specification the node was supposed to schedule.
+  /// \param status Status of the prepare response.
+  /// \param void
+  void MarkPreparePhaseDone(const ClientID &node_id,
+                            std::shared_ptr<BundleSpecification> bundle,
+                            const Status &status);
+
+  /// Used to know if all prepare requests are returend.
+  ///
+  /// \return True if all prepare requests are returned
+  bool IsAllPrepareRequestReturned() const;
+
+  /// Used to know if the prepare phase succeed.
+  ///
+  /// \return True if all prepare requests succeed.
+  bool IsPreparePhaseSucceed() const;
+
+  /// Return a placement group this status tracker is associated with.
+  ///
+  /// \return The placement group of this lease status tracker is tracking.
   const std::shared_ptr<GcsPlacementGroup> &GetPlacementGroup() const;
-  const std::vector<std::shared_ptr<BundleSpecification>> &GetUnplacedBundles() const;
-  const std::shared_ptr<BundleLocations> &GetBundleLocations() const;
+
+  /// Return bundles that should be scheduled.
+  ///
+  /// \return List of bundle specification that are supposed to be scheduled.
+  const std::vector<std::shared_ptr<BundleSpecification>> &GetBundlesToSchedule() const;
+
+  /// This method returns bundle locations that succeed to prepare resources.
+  ///
+  /// \return Location of bundles that succeed to prepare resources on a node.
+  const std::shared_ptr<BundleLocations> &GetPreparedBundleLocations() const;
+
+  /// Return the leasing state.
+  ///
+  /// \return Leasing state.
   const LeasingState GetLeasingState() const;
+
+  /// Mark that this leasing is cancelled.
+  ///
+  /// \return void.
   void MarkPlacementGroupScheduleCancelled();
 
+  /// Mark that the commit phase is started.
+  /// There's no need to mark commit phase is done because in that case, we won't need the
+  /// status tracker anymore.
+  void MarkCommitPhaseStarted();
+
  private:
   /// Method to update leasing states.
   ///
   /// \param leasing_state The state to update.
   /// \return True if succeeds to update. False otherwise.
   bool UpdateLeasingState(LeasingState leasing_state);
+
   /// Placement group of which this leasing context is associated with.
   std::shared_ptr<GcsPlacementGroup> placement_group_;
-  /// Location of bundles that lease requests were sent.
-  /// If schedule success, the decision will be set as schedule_map[bundles[pos]]
+
+  /// Location of bundles that prepare requests were sent.
+  /// If prepare succeeds, the decision will be set as schedule_map[bundles[pos]]
   /// else will be set ClientID::Nil().
-  std::shared_ptr<BundleLocations> bundle_locations_;
+  std::shared_ptr<BundleLocations> preparing_bundle_locations_;
+
   /// Number of lease requests that are returned.
-  size_t returned_count_ = 0;
+  size_t prepare_request_returned_count_ = 0;
```

Comment:
/// Number of lease requests that are returned. --- Comment need to be modified?